tools/wptrunner/wptrunner/browsers/base.py

```diff
@@ -275,3 +280,117 @@ def __call__(self, line):
         self.logger.process_output(self.pid,
                                    line.decode("utf8", "replace"),
                                    command=" ".join(self.command) if self.command else "")
+
+
+class WebDriverBrowser(Browser):
+    __metaclass__ = ABCMeta
+
+    default_base_path = "/"
+
+    def __init__(self, logger, binary, webdriver_binary, webdriver_args=None,
+                 host="127.0.0.1", port=None, base_path="", env=None, **kwargs):
+        super().__init__(logger)
+
+        if webdriver_binary is None:
+            raise ValueError("WebDriver server binary must be given "
+                             "to --webdriver-binary argument")
+
+        self.logger = logger
+        self.binary = binary
+        self.webdriver_binary = webdriver_binary
+
+        self.host = host
+        self._port = port
+
+        if base_path == "":
+            self.base_path = self.default_base_path
+        else:
+            self.base_path = base_path
+        self.env = os.environ.copy() if env is None else env
+        self._args = webdriver_args if webdriver_args is not None else []
+
+        self.url = "http://%s:%s%s" % (self.host, self.port, self.base_path)
+
+        self._output_handler = None
+        self._cmd = None
+        self._proc = None
+
+    def make_command(self):
+        """Returns the full command for starting the server process as a list."""
+        return [self.webdriver_binary] + self._args
+
+    def start(self, group_metadata, **kwargs):
+        try:
+            self._run_server(group_metadata, **kwargs)
+        except KeyboardInterrupt:
+            self.stop()
+
+    def create_output_handler(self, cmd):
```

Comment:
Is there some way in Python to mark that a method will be overridden? If not for that, this method would be a bit pointless.